assets/js/filters/block-list-block.js

```diff
@@ -1,10 +1,40 @@
 /**
  * External dependencies
  */
-import { withDefaultAttributes } from '@woocommerce/block-hocs';
-
+import { createHigherOrderComponent } from '@wordpress/compose';
+const { getBlockType } = wp.blocks;
 const { addFilter } = wp.hooks;
 
+/**
+ * withDefaultAttributes HOC for editor.BlockListBlock. This sets defaults
+ * when a block is loaded.
+ *
+ *  @param	object BlockListBlock The BlockListBlock element.
+ */
+const withDefaultAttributes = createHigherOrderComponent(
+	( BlockListBlock ) => {
+		return function( props ) {
+			const blockType = getBlockType( props.block.name );
+
+			if ( blockType.name.startsWith( 'woocommerce/' ) ) {
+				Object.keys( blockType.attributes ).map( ( key ) => {
+					if (
+						typeof props.attributes[ key ] === 'undefined' &&
+						typeof blockType.defaults !== 'undefined' &&
+						typeof blockType.defaults[ key ] !== 'undefined'
+					) {
+						props.attributes[ key ] = blockType.defaults[ key ];
```

Comment:
I just realized, we shouldn't be directly mutating props. While on the surface this works well, I'm concerned about unexpected bugs surfacing in the future because of this mutation.  Instead, I think we should create a new attributes object based off the old and feed that to an attributes prop. So essentially:

```js
return function( { attributes, ...props } ) {
   //below would be inside your blockType.name conditional block ...basically creating a new object.
   attributes = Object.assign( {}, attributes );
   // then you'd have your logic replacing undefined keys if needed.
  return <BlockListBlock attributes={ attributes } { ...props } />
}
```